app/services/deploy_service.rb

```diff
@@ -28,9 +31,22 @@ def self.create_github_pull_request(deploy_strategy)
         group_by(&:login).map{|k,v| [v.size, k] }.sort.reverse.map(&:last)
       assignee = sorted_logins.detect { |l| client.check_assignee(repo, l) }
       client.add_assignees(repo, pr.number, assignee) if assignee
+      pr_info = { html_url: pr.html_url, repo: repo }
     rescue Octokit::UnprocessableEntity
       # PR already exists
     end    
+    pr_info
+  end
+
+  def self.post_slack_notification(deploy_strategy, pr_info)
+    slack_channel = deploy_strategy.arguments['slack_channel']
+    return unless slack_channel
+    slack_token = deploy_strategy.profile&.environment&.fetch('SLACK_API_TOKEN')
```

Comment:
I see that the api token is on the `deploy_strategy` I wonder if that means we need to set a token/channel for every repo? I wonder if there are some sensible defaults we can use?

Reply:
The API token is actually on the `Profile` associated with the deploy strategy. `Profile`s are Horizon's model for "identity" and any associated secrets. This has been useful, e.g., because it has to connect to different git repos with different identities--sometimes even for the different stages of a single project, such as when `master` is on Github but `staging` is based on Heroku. This means we should only need to set the token in one place.

The channel setting _is_ on each deploy strategy though. This was my earlier suggestion, so that different repos can report their deploys to different channels.